controllers/users.js

```diff
@@ -63,3 +87,42 @@ module.exports.logout = (req, res) => {
 	req.flash('success', 'Success logout!');
 	res.redirect('/offers');
 };
+
+module.exports.profilePage = async (req, res) => {
+	const queryData = await UserInformation.findOne( {'username': `${(req.params.name) ? req.params.name : req.user.username}`});
+	res.render('users/profile', { queryData });
+};
+
+module.exports.editPage = async (req, res) => {
+	res.render('users/edit')
+}
+
+module.exports.createEditPage = async (req, res) => {
+
+	const filter = { username: req.user.username };
+	const option = { new: true, upsert: true }
+	const userinfo = {
+		username: req.user.username,
+		email: req.user.email,
+		firstname: req.body.firstname,
+		surname: req.body.surname,
+		phone: req.body.phone,
+		address1: req.body.address1,
+		address2: req.body.address2,
+		postcode: req.body.postcode,
+		area: req.body.area,
+		country: req.body.country,
+		state: req.body.state,
+		description: req.body.description,
+		details: req.body.details	
+	};
+
+
+	//validate inputs "WILL FIX LATER IF NECESSARY".
+	//const userSchema = await createUserSchema.validateAsync(userinfo);
+
+	//Update DB.
+	await UserInformation.findOneAndUpdate(filter, userinfo, option);

```

Comment:
Ja, validation är nog fint! då har den koll så vi inte sätter in extra keys (så man inte behöver hämta ut alla keys).